### PR TITLE
docs - consolidate database connection settings

### PR DESCRIPTION
**Pull request by jeff-bruemmer**
Updated the docs for several databases, and want to get feedback and get this PR over the line before proceeding with the other edits.

Opted not to consolidate all database settings into a single page. Each database page should list all of its settings. This approach is more of a hassle, maintenance-wise, but the reader's experience should justify the repetition of setting definitions.

### CODE REVIEW

File: docs/databases/connections/snowflake.md [hunk 1]
```diff
@@ -6,9 +6,47 @@ redirect_from:
...
+
+### Display name
+
+What you want Metabase to call the database in its user interface.
+
+### Account name
+
+Enter your Account ID with the region that your Snowflake cluster is running on. E.g., `xxxxxxxxx.us-east-3.aws`.
+
+This field requires the alphanumeric account ID _with_ the region that your Snowflake cluster is running on. For example, if you're running Snowflake on AWS and your account URL is `https://az12345.ca-central-1.snowflakecomputing.com`, then the `Account` would be `az12345.ca-central-1.aws` (note the `.aws` suffix). There are some regions that don't need this suffix, so please [refer to the official Snowflake documentation](https://docs.snowflake.com/en/user-guide/admin-account-identifier.html#locator-formats-by-cloud-platform-and-region) for this
```
Comment (nllho) [inline docs/databases/connections/snowflake.md:23]:
```suggestion
Enter your Snowflake [Account Identifier](https://docs.snowflake.com/en/user-guide/admin-account-identifier.html). For example, if you're running Snowflake on AWS and your account URL is `https://az12345.ca-central-1.snowflakecomputing.com`:
- `<account_identifier>`: `az12345.ca-central-1`.
- `<cloud_platform>`: `aws`.

You'd enter `az12345.ca-central-1.aws` as the account name in Metabase.

> Not all regions require the cloud platform identifier. If you are in `us-west-2`, you would enter `az12345` as the account name. For the requirements per region, see [the official Snowflake's documentation](https://docs.snowflake.com/en/user-guide/admin-account-identifier.html#non-vps-account-locator-formats-by-cloud-platform-and-region).
```

Might as well update this, it was in the backlog already. For reference: https://discourse.metabase.com/t/time-out-snowflake-connection/23295.

File: docs/databases/connections/snowflake.md [hunk 2]
```diff
@@ -17,24 +55,79 @@ Here you can specify which schemas you want to sync and scan. Options are:
...
-- If you have **All except...** set, and enter the string `b*`, you'll just sync foo.
+- If you have **Only these...** set, and enter the string `B*`, you'll sync with bar and baz.
+- If you have **All except...** set, and enter the string `B*`, you'll just sync foo.
 
 Note that only the `*` wildcard is supported; you can't use other special characters or regexes.
 
-## Snowflake gotchas
+### Role (optional)
+
+Specify a role to override the database user's default role. For example, if the database user is `REPORTER` with default role `REPORTER`, but the user also has access to role `REPORTERPRODUCT`, then filling in `REPORTERPRODUCT` in the `Role` field will ensure that the `REPORTERPRODUCT` role is used instead of the user's default `REPORTER` role.
```
Comment (nllho) [inline docs/databases/connections/snowflake.md:72]:
```suggestion
Specify a role to override the database user's default role. For example, if the database user `METABASE` has the roles:
- Default role `APPLICATION`.
- Additional role `ANALYTICS`.

You can enter `ANALYTICS` in the Role field to ensure that the `METABASE` user connects to Snowflake using the `ANALYTICS` role by default.
```

For ease of reading, and to make the example more relevant to a db connection.

File: docs/databases/connecting.md [hunk 3]
```diff
@@ -26,7 +26,7 @@ The databases listed below have official drivers maintained by the Metabase team
 - Druid
 - [Google Analytics](./connections/google-analytics.md)
 - H2
```
Comment (nllho) [inline docs/databases/connecting.md:28]: Don't forget to update the links!

**Review by nllho — APPROVED** [review]
Thanks for going through all of these! 

A few suggestions -- mostly stuff I've seen confuse people in tickets / from my experience, but nothing blocking if we want to add those another time.

File: docs/databases/connections/mysql.md [hunk 4]
```diff
@@ -6,16 +6,99 @@ redirect_from:
...
+
+You can paste your server's SSL certification chain.
+
+### Use an SSH tunnel
+
+See our [guide to SSH tunneling](../ssh-tunnel.md).
+
+### Unfold JSON Columns
+
+In some databases, Metabase can unfold JSON columns into component fields to yield a table where each JSON key becomes a column. JSON unfolding is on by default, but you can turn off JSON folding if performance is slow.
```
[the remark this answers is not in the document]
Comment (jeff-bruemmer) [reply docs/databases/connections/mysql.md:45]: Going to leave this as is, as it matches the in-product text.

File: docs/databases/connections/mysql.md [hunk 5]
```diff
@@ -6,16 +6,99 @@ redirect_from:
...
+Options are:
+
+- **Use instance default (TTL)**. TTL is time to live, meaning how long the cache remains valid before Metabase should run the query again.
+- **Custom**.
+
+If you are on a paid plan, you can also set cache duration per questions. See [Advanced caching controls](../../configuring-metabase/caching.md#advanced-caching-controls).
 
 ## Connecting to MySQL 8+ servers
 
 Metabase uses the MariaDB connector to connect to MariaDB and MySQL servers. The MariaDB connector does not currently support MySQL 8's default authentication plugin, so in order to connect, you'll need to change the plugin used by the Metabase user to `mysql_native_password`: `ALTER USER 'metabase'@'%' IDENTIFIED WITH mysql_native_password BY 'thepassword';`
```
[the remark this answers is not in the document]
Comment (jeff-bruemmer) [reply docs/databases/connections/mysql.md:99]: Yeah, code block should fix this.

**Pull request merged**